cmd/proxy/udp_proxy.go

```diff
@@ -92,7 +114,28 @@ func (proxy *UDPProxy) replyLoop(proxyConn *net.UDPConn, clientAddr *net.UDPAddr
 			return
 		}
 		for i := 0; i != read; {
-			written, err := proxy.listener.WriteToUDP(readBuf[i:read], clientAddr)
+			written := 0
+
+			if srcAddr != nil {
+				// set the src IP of the response to match the DST of the
+				// initial client packet.
+				var oob []byte //= nil
+
+				if srcAddr.To4() != nil {
+					cm := new(ipv4.ControlMessage)
```

Comment:
This seems expensive here. Can we re-use control messages?
Even then we'll be allocating for `Marshal` which is also not great.

Reply:
If I'm parsing the go documentation correctly (which I'm likely not 😅 it's a normal language for me) I think `cm` is both read and written by `WriteMsgUDP`, so reusing it mightn't be appropriate as the state of the other `cm` members might have been mutated.

Is there anything else in the change I could modify to get it across the line performance wise? I appreciate the increased cost to processing UDP clients but it seem necessary for UDP correctness.

Reply:
`cm` is only being used to marshal a message nothing shared between uses from what I can tell.
Each call to `Marshal()` is a new byte slice.

Reply:
Sorry I’m on the same page now - working through a fix and addressing regressions now